### ISSUE TITLE
Should Discrete Sampler add a softmax in `p_1t`?

### ISSUE THREAD

**Issue by ixsluo** [facebookresearch/flow_matching#46]
In discrete solver, it directly use the `self.model`'s output:

https://github.com/facebookresearch/flow_matching/blob/47c439602a360ef7d2ec2b9a521b4dda5f335112/flow_matching/solver/discrete_solver.py#L198-L200

to create a categorical `x_1` by `torch.multinomial`:

https://github.com/facebookresearch/flow_matching/blob/47c439602a360ef7d2ec2b9a521b4dda5f335112/flow_matching/utils/categorical_sampler.py#L21-L23

However, in the [examples/2d_discrete_flow_matching.ipynb](https://github.com/facebookresearch/flow_matching/blob/47c439602a360ef7d2ec2b9a521b4dda5f335112/examples/2d_discrete_flow_matching.ipynb), the model output is not a probability.

Besides, MixturePathGeneralizedKL do assume the model output is not softmax:

https://github.com/facebookresearch/flow_matching/blob/47c439602a360ef7d2ec2b9a521b4dda5f335112/flow_matching/loss/generalized_loss.py#L51-L52

Does it need a softmax in getting `p_1t` in solver?

**Comment by helibenhamu** [comment]
The `torch.multinomial` function handles both logits and probabilities. See documentation here: https://docs.pytorch.org/docs/stable/generated/torch.multinomial.html
